src/ScottPlot4/ScottPlot/SnapLogic/Custom.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace ScottPlot.SnapLogic;
+
+/// <summary>
+/// Create a snap object from a custom function
+/// </summary>
+public class Custom : ISnap
```

Comment:
Not sure if this is necessary compared to people implementing the interface themselves? Feel free to discard if you've already discussed this.

Reply:
It's a bit of a hack, but it makes the cookbook possible

```cs
public void ExecuteRecipe(Plot plt)
{
    double CustomSnapFunction(double value)
    {
        // multiple of 3 between 0 and 50
        if (value < 0) return 0;
        else if (value > 50) return 50;
        else return (int)Math.Round(value / 3) * 3;
    }

    // different snap sytems can be created and customized 
    var SnapDisabled = new ScottPlot.SnapLogic.Smooth();
    var SnapCustom = new SnapLogic.Custom(CustomSnapFunction);

    plt.AddSignal(DataGen.Sin(51, mult: 5));
    plt.AddSignal(DataGen.Cos(51, mult: 5));

    var vLine = plt.AddVerticalLine(30);
    vLine.DragEnabled = true;
    vLine.DragSnap = new ScottPlot.SnapLogic.Independent2D(SnapCustom, SnapDisabled);
}
```